scripts/generate_vif/generate_vif.py

```diff
@@ -0,0 +1,259 @@
+            pdos_info.get(constants.FR_SWAP_REQD_TYPE_C_CURRENT_AS_INITIAL_SOURCE, 0))})
+    add_element_to_xml(xml_ele, constants.PD_POWER_AS_SINK, f'{snk_max_power} mW',
+                       {constants.VALUE: str(snk_max_power)})
+
+
+def parse_and_add_sink_pdo_to_xml(xml_ele, pdo_value, pdos_info):
+    power_mw = 0
+    xml_ele = add_element_to_xml(xml_ele, constants.SINK_PDO)
+    pdo_type = get_pdo_type(pdo_value)
+    if pdo_type == constants.PDO_TYPE_FIXED:
+        current = pdo_value & 0x3ff
+        current_ma = current * 10
+        voltage = (pdo_value >> 10) & 0x3ff
+        voltage_mv = voltage * 50
+        power_mw = (current_ma * voltage_mv) // 1000
+        pdos_info[constants.HIGHER_CAPABILITY_SET] = pdo_value & (1 << 28)
+        pdos_info[constants.FR_SWAP_REQD_TYPE_C_CURRENT_AS_INITIAL_SOURCE] = pdo_value & (3 << 23)
+        add_element_to_xml(xml_ele, constants.SINK_PDO_VOLTAGE, f'{voltage_mv} mV',
+                           {constants.VALUE: str(voltage)})
+        add_element_to_xml(xml_ele, constants.SINK_PDO_OP_CURRENT,
+                           f'{current_ma} mA',
+                           {constants.VALUE: str(current)})
+    elif pdo_type == constants.PDO_TYPE_BATTERY:
+        max_voltage = (pdo_value >> 20) & 0x3ff
+        max_voltage_mv = max_voltage * 50
+        min_voltage = (pdo_value >> 10) & 0x3ff
+        min_voltage_mv = min_voltage * 50
+        power = pdo_value & 0x3ff
+        power_mw = power * 250
+        add_element_to_xml(xml_ele, constants.SINK_PDO_MIN_VOLTAGE,
+                           f'{min_voltage_mv} mV',
+                           {constants.VALUE: str(min_voltage)})
+        add_element_to_xml(xml_ele, constants.SINK_PDO_MAX_VOLTAGE,
+                           f'{max_voltage_mv} mV',
+                           {constants.VALUE: str(max_voltage)})
+        add_element_to_xml(xml_ele, constants.SINK_PDO_OP_POWER, f'{power_mw} mW',
+                           {constants.VALUE: str(power)})
+    elif pdo_type == constants.PDO_TYPE_VARIABLE:
+        max_voltage = (pdo_value >> 20) & 0x3ff
+        max_voltage_mv = max_voltage * 50
+        min_voltage = (pdo_value >> 10) & 0x3ff
+        min_voltage_mv = min_voltage * 50
+        current = pdo_value & 0x3ff
+        current_ma = current * 10
+        power_mw = (current_ma * max_voltage_mv) // 1000
+        add_element_to_xml(xml_ele, constants.SINK_PDO_MIN_VOLTAGE,
+                           f'{min_voltage_mv} mV',
+                           {constants.VALUE: str(min_voltage)})
+        add_element_to_xml(xml_ele, constants.SINK_PDO_MAX_VOLTAGE,
+                           f'{max_voltage_mv} mV',
+                           {constants.VALUE: str(max_voltage)})
+        add_element_to_xml(xml_ele, constants.SINK_PDO_OP_CURRENT,
+                           f'{current_ma} mA',
+                           {constants.VALUE: str(current)})
+    elif pdo_type == constants.PDO_TYPE_AUGUMENTED:
+        pps = (pdo_value >> 28) & 3
```

Comment:
nit: I'd use 0x03 other than 3... just for coherency, all other bitmask are against a hex

Reply:
Updated it with hex(3)